docs/proposals/ThumbnailGallery/README.md

```diff
@@ -0,0 +1,60 @@
+# Component Name
+
+The `ThumbnailGallery` component will accept a list of attachments for images
+and display them as thumbnails. This would also have an option to show the
+images in a LightBox component on click of those images.
+
+## Accessibility
+
+Assistive technology should be able to see the images and know the image name
+based on the `name` property of each attachment.
+
+## Responsiveness & Mobile
+
+This component show maximum number of images possible in a line and show the
+count of images left to show.
+
+## Wireframe
+
+[mockup](mockup.png)
+
+## Interface
+
+```
+<ThumbnailGallery
+    attachments={[{
+            name: "filename.png",
```

Comment:
We have a `FileUpload` interface in `packages/components/src/InputFile/InputFile.tsx` that's probably good to use for these.

Reply:
I was looking at it for the other PR, and it looks like `FileUpload` has the progress property which in this case would always be 1. So thought of having another interface specific for Gallery that does not require progress key but would be passing it as 1 when it is sending it to FormatFile.

Reply:
I'll address this in the review body - I think we can find a way to support both types.